### PR TITLE
adding module that implements filtering tesseract output by confidenc…

### PR DESCRIPTION
**Pull request by prockwood**
## Description
tess_conf_filter.py
I'm not intending to merge this, hence the draft PR. Just wanted to put it up where everyone can take a look, and invite any feedback or thoughts.

There's two functions:
    - one to turn images into tessaract's ocr data, which includes confidence levels at the word level.
    - one to filter out words that are below a specified confidence level, and return what's leftover as a string.

My initial motivation was to find an automated way to detect images that were not textual or amenable to ocr, so that we weren't storing a lot of garbage. This is a bit harder than I expected. With confidence at the word level, we could take the mean of confidence levels, and reject pages below a threshold. This might reject stuff we would like to keep, a map or photo with a caption for instance. As it stands this does offer a way to limit the amount of garbage text we store on mongodb.

### CODE REVIEW

**Review by lucchesia7 — APPROVED** [review]

File: tess_conf_filter.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ocr, extracting words and confidence levels per word, then filters out words
+below a specified confidence level.
+"""
+
+from PIL import Image
+import pytesseract
+import pdf2image
+import time
+import re
+import pandas as pd
```
Comment (prockwood) [inline tess_conf_filter.py:12]: I forgot to mention in the description: this uses pandas, which is not currently in the requirements.txt.

**Review by madina-iz — APPROVED** [review]
Great potential functionality! Very good explanation of the addition as well.
I was thinking about the potential issue you mentioned (losing the text that we actually want to keep) and wonder if it would make sense to set the threshold to a fixed level rather than the mean. I am just curious, that's all.

**Review by yalyakoob — APPROVED** [review]
good work!

**Comment by engegreg** [comment]
Looks good, if you intend to merge this in the future make sure to remove the hardcoded file id from the name __main__ call. Other than that this streamlines a lot of things in the data pipeline.

**Review by galtman5 — APPROVED** [review]

**Review by jhebert1994 — APPROVED** [review]
Looks good, think this could have great potential in future updates.

**Comment by prockwood** [comment]
replaced by feature/tesseract_conf_flag

**Pull request closed without merging**